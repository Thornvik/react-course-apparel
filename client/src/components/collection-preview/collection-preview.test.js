import React from "react";
import { shallow } from "enzyme";
import { CollectionPreview } from "./collection-preview.component";

describe("collection-preview", () => {
  let wrapper;
  let mockHistory;
  let mockMatch;
  const mockRouteName = "hats";

  beforeEach(() => {
    mockHistory = {
      push: jest.fn(),
    };

    mockMatch = {
      path: "/shop",
    };

    const mockProps = {
      match: mockMatch,
      history: mockHistory,
      routeName: mockRouteName,
      title: "hats",
      items: [],
    };

    wrapper = shallow(<CollectionPreview {...mockProps} />);
  });

  it("should render collection preview", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call history.push with correct string when title is clicked", () => {
    wrapper.find(".title").simulate("click");

    expect(mockHistory.push).toHaveBeenCalledWith(
      `${mockMatch.path}/${mockRouteName}`
    );
  });
});
