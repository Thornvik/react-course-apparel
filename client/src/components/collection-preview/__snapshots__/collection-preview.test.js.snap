// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collection-preview should render collection preview 1`] = `ShallowWrapper {}`;
